Anagram.java

```diff
@@ -26,9 +27,58 @@ public static void main(String args[]) {
 		System.out.println(pass ? "test passed" : "test Failed");
 	}  
 
+
+	public static String onlyLowerCaseString(String str){
+		// function that get a string input and return him with only lower case and without spaces or punctuation marks.
+		String newString = "";
+		char newLettar;
+		for(int i = 0; i<str.length(); i++){
+			if (((int)str.charAt(i)) < 97 || ((int)str.charAt(i))>122) {
+				if (((int)str.charAt(i)) > 64 && (int)str.charAt(i) < 96) {
+					newLettar = (char)(((int)str.charAt(i))+32);
+					newString = newString + newLettar;
+				}
+			}
+			else{
+				newString = newString + str.charAt(i);
+			}
+		}
+			return newString;
+	} 
+
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
 		// Replace the following statement with your code
+		
+		str1 = onlyLowerCaseString(str1);
+		str2 = onlyLowerCaseString(str2);
+
+		if (str1.length() != str2.length()) {
+			return false;
+		}
+
+		int i = 0 , j=0;;
+		boolean checkChar = true;
+		boolean checkChar2 = false;
+		while (i<str1.length() && checkChar==true) {
+			while (j<str2.length() && checkChar2==false) {
+				if (str1.charAt(i) == str2.charAt(j)) {
+					checkChar2 = true;
+				}
+				j++;
+			}
+			if (checkChar2==false) {
+				checkChar = false;
+			}
+			else{
+				i++;
+				checkChar2 = false;
+				j=0;
+			}
```

Comment:
1. looks like for by definition to me.. 
2. use indicative names please
3. spacing
4. `while (i < str1.length() && checkChar)`, `while (j < str2.length() && !checkChar2)`, `if (!checkChar2)` 
5. `str1.indexOf(str2.charAt(j))`